### ISSUE TITLE
v10 disabled buttons need usage documentation

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#1701]
### Summary

Our disabled state buttons meet [WCAG 2.1 spcifications for color contrast on disabled components](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html), but a disabled user interface element could have varying significance depending on _why_ it's disabled. The example the specs give are 

1. The button is disabled because the user hasn't done something yet. So the button is communicating to the user that there is an unmet requirement of some kind
2. The button is disabled because it has no significance in the current layout or task.

In the first case the information conveyed by the button should be accessible to all of our users and a greater contrast (3:1 or better) should be used.

In the second case our specified disabled buttons are sufficient since the button ultimately doesn't matter.

For an example of the kind of documentation needed I'm thinking something like this from the [gov.uk design system](https://design-system.service.gov.uk/components/button/):
_"Disabled buttons have poor contrast and can confuse some users, so avoid them if possible. Only use disabled buttons if research shows it makes the user interface easier to understand."_

**Comment by paschalidi** [comment]
@dakahn hey 👋 thats a great idea! I will try to open a PR the next days and you could give me feedback on how you imagine this to be. 

Thanks for raising this!

**Comment by dakahn** [comment]
Closing this. The first point is something we can address in upcoming patterns and training work and a general note about the use of disabled buttons to communicate unmet requirements being bad UX can make it into our usage/implementation docs
